runtime/onert/api/nnfw/src/nnfw_api_internal.cc

```diff
@@ -1545,6 +1545,53 @@ NNFW_STATUS nnfw_session::train_set_output(uint32_t index, NNFW_TYPE /*type*/, v
   return NNFW_STATUS_NO_ERROR;
 }
 
+NNFW_STATUS nnfw_session::nnfw_train_disable_node_update(uint32_t op_index)
+{
+  if (!isStateModelLoaded() || isStatePreparedTraining())
+  {
+    std::cerr << "Error during nnfw_session::nnfw_train_disable_node_update: invalid session state"
+              << std::endl;
+    return NNFW_STATUS_INVALID_STATE;
+  }
+
+  if (op_index >= primary_subgraph()->operations().size())
+  {
+    std::cerr << "Error during nnfw_session::nnfw_train_disable_node_update: provided op_index="
+              << op_index << " is out of operators range" << std::endl;
+    return NNFW_STATUS_INVALID_STATE;
+  }
+
+  const auto ir_op_index = onert::ir::OperationIndex{op_index};
+  auto &options = _coptions[0];
```

Comment:
How about using `TrainingInfo` instead of _coptions?

Reply:
@jyoungyun I like this idea, introduced ;)